include/zephyr/kernel.h

```diff
@@ -2881,7 +2881,7 @@ __syscall int k_stack_pop(struct k_stack *stack, stack_data_t *data,
  */
 #define K_STACK_DEFINE(name, stack_num_entries)                \
 	stack_data_t __noinit                                  \
-		_k_stack_buf_##name[stack_num_entries];        \
+		_k_stack_buf_##name[(stack_num_entries)];      \
```

Comment:
Similarly this isn't a rule violation: you can't "escape" a nested context like square brackets with an operator expression.  Anything that binds tighter than the '[' or ']' tokens would be a parse error.  It's the same reason we don't need to parenthesize macro arguments that are passed directly to single-argument functions, e.g. `#define WRAPPED_A(x) function_a(x)` is fine too.